src/Unic.ErrorManager.Website/sitecore modules/Web/Error Manager/404.aspx.cs

```diff
@@ -31,7 +31,10 @@ public partial class _404 : BaseError
         protected override void OnLoad(EventArgs e)
         {
             // set properties
-            base.SettingsKey = "ItemNotFoundUrl";
+            var isMedia = Context.Items.Contains("IsMedia")
+                ? Convert.ToBoolean(Context.Items["IsMedia"])
+                : Sitecore.Web.WebUtil.GetItemsBool("IsMedia", false);
```

Comment:
This seems to be redundant with the previous line. Did you want to consider the query string parameter here?

Reply:
yes, it will be set by MediaRequestHandler when UseServerSideRedirect is disabled